### PR TITLE
Create working search in scaffoldvuer

### PR DESCRIPTION
**Pull request by Tehsurfer**
Can test the search in App.vue and also in mapintegratedvuer

https://github.com/ABI-Software/mapintegratedvuer/pull/157

### CODE REVIEW

File: src/components/ScaffoldVuer.vue [hunk 1]
```diff
@@ -484,6 +485,8 @@ export default {
         external: false,
       },
       fileFormat: "metadata",
+      searchIndex: undefined,
+      searchTooltipActive: false,
```
Comment (alan-wu) [inline src/components/ScaffoldVuer.vue:489]: This is not used. Remove this please.

File: src/App.vue [hunk 2]
```diff
@@ -263,7 +274,7 @@ import { ScaffoldVuer } from "./components/index.js";
 import DropZone from "./components/DropZone.vue";
 import ModelsTable from "./components/ModelsTable.vue";
 import Vue from "vue";
-import { Button, Col, Icon, Input, InputNumber, Popover, Row, Switch } from "element-ui";
+import { Button, Col, Icon, Input, InputNumber, Popover, Row, Switch, stopAutoTumble, Autocomplete } from "element-ui";
```
Comment (alan-wu) [inline src/App.vue:277]: Typo with stopAutoTumble here. It should not be here.

File: src/components/ScaffoldVuer.vue [hunk 3]
```diff
@@ -484,6 +485,8 @@ export default {
         external: false,
       },
       fileFormat: "metadata",
+      searchIndex: undefined,
```
Comment (alan-wu) [inline src/components/ScaffoldVuer.vue:488]: searchIndex is not reactive, preferably remove this and move it over to beforeCreate.

File: src/scripts/search.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    'label',
+    'models',
+    'source'
+];
+
+//==============================================================================
+
+export class SearchIndex
+{
+    constructor()
```
Comment (alan-wu) [inline src/scripts/search.js:40]: Using constructor and class may cause problem on mapintegratedvuer.

File: src/scripts/utilities.js [hunk 5]
```diff
@@ -35,3 +35,38 @@ export const findObjectsWithNames = (rootRegion, names, regionPath, transverse)
   }
   return targetObjects;
 }
+
+export const getAllGroupNames = (scene) => {
+  let groupNames = [];
+  scene.forEachGeometry((geometry) => {
```
Comment (alan-wu) [inline src/scripts/utilities.js:41]: More efficient to =>

`scene.getRootRegion().getAllObjects(transverse);`

then iterates through that.

File: src/scripts/utilities.js [hunk 6]
```diff
@@ -35,3 +35,38 @@ export const findObjectsWithNames = (rootRegion, names, regionPath, transverse)
...
+  scene.forEachLine((line) => {
+    if (line.groupName && !groupNames.includes(line.groupName))
+      groupNames.push(line.groupName);
+  })
+  return groupNames;
+}
+
+export const getAllObjects = (scene) => {
+  let objects = [];
+  scene.forEachGeometry((geometry) => {
```
Comment (alan-wu) [inline src/scripts/utilities.js:58]: More efficient to =>

`scene.getRootRegion().getAllObjects(transverse)`

**Comment by Tehsurfer** [comment]
Note:

I will switch the search from highlighting to selecting an active region.

Also will move the search function from scaffoldvuer to mapintegratedvuer

**Review by alan-wu — CHANGES_REQUESTED** [review]

File: src/scripts/search.js (hunk 4 above)
Comment (Tehsurfer) [reply src/scripts/search.js:40]: Should I switch to function?

**Review by alan-wu — APPROVED** [review]

**Pull request merged**